core-curriculum/knowledge/linux.json

```diff
@@ -0,0 +1,225 @@
+            "tags":["required-for-junior"],
+            "dependencies":["basic"],
+            "abilities": [
+                "Log into a Linux box using SSH.",
+                "Generating ssh private keys.",
+                "Setting up a port forward on an SSH tunnel.",
+                "Execute commands directly on remote host.",
+                "Understand the limitations of SSH (e.g.: what happens if I start a long running migration in an SSH session and my connection breaks)"
+            ],
+            "resources":[
+                {
+                    "name":"Digitalocean: SSH Essentials",
+                    "type":"Article",
+                    "url":"https://www.digitalocean.com/community/tutorials/ssh-essentials-working-with-ssh-servers-clients-and-keys"
+                }
+            ]
+        },
+        {
+            "id": "bashtricks",
+            "name": "Using Bash",
+            "description":"The most common shell on Linux is bash. Bash and its associated utilities probably deserve a knowledge graph on their own. However even with some basic knowledge about subjects such as piping and output redirection, and where to find the manual, it can quickly become a powerful tool for any user.",
+            "tags":[],
+            "dependencies":["basic"],
+            "abilities": [
+                "Redirect output of a program to a file and/or another program.",
+                "Redirect stdout to stderr and vice versa.",
+                "Write a basic bash script.",
+                "Basic data manipulation, e.g.: search/replace, remove a column from a csv, extract values from json."
+            ],
+            "resources":[
+                {
+                    "name": "Linuxjourney lesson - Text-Fu",
+                    "type": "article",
+                    "url": "https://linuxjourney.com/lesson/stdout-standard-out-redirect"
+                }
+            ]
+        },
+        {
+            "id": "networking-advanced",
+            "name": "Networking II",
+            "description":"Packet inspection, performance measurement and advanced firewalling/routing setups (e.g.: using linux for a home router).",
+            "tags":[],
+            "dependencies":["networking"],
+            "abilities": [
+                "Set up a linux machine as a router/gateway for a network."
+            ],
+            "resources":[]
+        },
+        {
+            "id": "packagemanagement",
+            "name": "Package management I",
+            "description":"There are almost as many different package managers as there are linux distributions. However they share many common operations, just with a different syntax. The goal of this topic is to get an understanding of the common operations: installing, removing, upgrading.",
+            "tags":["required-for-junior"],
+            "dependencies":["basic"],
+            "abilities": [],
```

Comment:
since usually you will encounter either Debian or Red Hat can we limit it to `apt` and `yum`/`dnf`?

Reply:
Seems like a good idea, maybe also add a link to the "package management rosetta" by Arch Linux (https://wiki.archlinux.org/index.php/Pacman/Rosetta ).
I was thinking of also including  Alpine btw, as many lightweight docker images are based on that.